code/modules/events/supermatter_surge.dm

```diff
@@ -0,0 +1,110 @@
+#define SURGE_DURATION_MAX 270 EVENT_SECONDS
+#define SURGE_SEVERITY_MIN 1
+#define SURGE_SEVERITY_MAX 4
+/// The amount of bullet energy we add for the duration of the SM surge
+#define SURGE_BULLET_ENERGY_ADDITION 5
+/// The amount of powerloss inhibition (energy retention) we add for the duration of the SM surge
+#define SURGE_BASE_POWERLOSS_INHIBITION 0.55
+/// The powerloss inhibition scaling based on surge severity
+#define SURGE_POWERLOSS_INHIBITION_MODIFIER 0.175
+/// The power generation scaling based on surge severity
+#define SURGE_POWER_GENERATION_MODIFIER 0.075
+/// The heat modifier scaling based on surge severity
+#define SURGE_HEAT_MODIFIER 0.25
+
+/**
+ * Supermatter Surge
+ *
+ * An engineering challenge event where the properties of the SM changes to be in a 'surge' of power.
+ * For the duration of the event a powerloss inhibition is added to nitrogen, causing the crystal to retain more of its internal energy.
+ * Heat modifier is lowered to generate some heat but not a high temp burn.
+ * Bullet energy from emitters is raised slightly to raise meV while turned on.
+ */
+
+/datum/round_event_control/supermatter_surge
+	name = "Supermatter Surge"
+	typepath = /datum/round_event/supermatter_surge
+	category = EVENT_CATEGORY_ENGINEERING
+	weight = 15
+	max_occurrences = 1
+	earliest_start = 20 MINUTES
+	description = "The supermatter will increase in power and heat by a random amount, and announce it."
+	min_wizard_trigger_potency = 4
+	max_wizard_trigger_potency = 7
+	admin_setup = list(
+		/datum/event_admin_setup/input_number/surge_spiciness,
+	)
+
+/datum/round_event_control/supermatter_surge/can_spawn_event(players_amt, allow_magic = FALSE)
+	. = ..()
+
+	if(SSjob.is_skeleton_engineering(crew_threshold = 3))
+		return FALSE
+
+/datum/round_event/supermatter_surge
+	announce_when = 4
+	end_when = SURGE_DURATION_MIN
+	/// How powerful is the supermatter surge going to be?
+	var/surge_class =  SURGE_SEVERITY_MIN
+	/// Typecasted reference to the supermatter chosen at event start
+	var/obj/machinery/power/supermatter_crystal/engine
+	/// Typecasted reference to the nitrogen properies in the SM chamber
+	var/datum/sm_gas/nitrogen/sm_gas
+
+/datum/event_admin_setup/input_number/surge_spiciness
+	input_text = "Set surge intensity. (Higher is more severe.)"
+	min_value = 1
+	max_value = 4
+
+/datum/event_admin_setup/input_number/surge_spiciness/prompt_admins()
+	default_value = rand(1, 4)
+	return ..()
+
+/datum/event_admin_setup/input_number/surge_spiciness/apply_to_event(datum/round_event/supermatter_surge/event)
+	event.surge_class = chosen_value
+
+/datum/round_event/supermatter_surge/setup()
+	engine = GLOB.main_supermatter_engine
+	if(isnull(engine))
+		stack_trace("SM surge event failed to find a supermatter engine!")
+		return
+
+	sm_gas = LAZYACCESS(engine.current_gas_behavior, /datum/gas/nitrogen)
+	if(isnull(sm_gas))
+		stack_trace("SM surge event failed to find gas properties for [engine].")
```

Comment:
If it's an intended & expected feature that this doesn't happen on nitrogenless engines we shouldn't stack trace here, and maybe even should announce the engine's resilience against the event in some way.
If you don't want to _run_ the event if there's no N2 gas then check for it earlier.

Reply:
Yeah I suppose it's not needed. If it can't get the gas mixture of the SM something else has already gone horribly wrong.

Reply:
As discussed, it's accessing the datum of nitrogen's properties, not the gases itself. It should be impossible unless the engine is deleted in the exact moment between the isnull check above it and accessing the datum.